File: app/src/main/java/com/example/weatherapp/DateConverter.java
package com.example.weatherapp;

import androidx.room.TypeConverter;

import java.time.LocalDateTime;
import java.sql.Date;
import java.time.format.DateTimeFormatter;

public class DateConverter {

    @TypeConverter
    public static LocalDateTime toDate(String timestamp) {
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy H:mm:ss");
        return LocalDateTime.parse(timestamp, formatter);
    }

    @TypeConverter
    public static String toTimestamp(LocalDateTime date) {
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy H:mm:ss");
        return date == null ? null : formatter.format(date);
    }
}

File: app/src/main/java/com/example/weatherapp/DateConverter.kt
package com.example.weatherapp

import androidx.room.TypeConverter
import java.time.LocalDateTime
import java.time.ZoneOffset
import java.time.format.DateTimeFormatter

object DateConverter {
    @JvmStatic
    @TypeConverter
    fun toDate(timestamp: Long): LocalDateTime {
        val formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy H:mm:ss")
        return LocalDateTime.parse(timestamp.toString(), formatter)
    }

    @JvmStatic
    @TypeConverter
    fun toTimestamp(date: LocalDateTime): Long {
        return date.toEpochSecond(ZoneOffset.UTC)
    }
}